semgrep/semgrep/commands/scan.py

```diff
@@ -533,6 +533,7 @@ def convert(
     hidden=True
     # help="WARNING: allow rules to run arbitrary code (pattern-where-python)",
 )
+@click.option("--dump-command-for-core", "-d", is_flag=True, hidden=True)
```

Comment:
For internal stuff might we be better served by an env var like `SEMGREP_DUMP_CORE_COMMAND`

Reply:
Env vars are so annoying for when you want to flip back and forth though

Reply:
yes but with this we need to send this flag all the way down increasing the interface of lots of functions for an internal thing